File: src/com/zypex/piflowapp/Graphics/Graph.java
package com.zypex.piflowapp.Graphics;

import javafx.css.Size;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.paint.Color;
import utils.math.Vector;

import java.util.ArrayList;
import java.util.List;

public class Graph {

    private final List<Vector> datapoints = new ArrayList<>();
    public Color color = Color.RED;
    public DataMode dataMode = DataMode.LINEAR;
    public SizeMode sizeMode = SizeMode.RELATIVE;

    public void drawSection(GraphicsContext gc){
        final double w = gc.getCanvas().getWidth();
        final double h = gc.getCanvas().getHeight();
    }

    public void addDatapoint(double x, double y){
        for(int i = 0; i < datapoints.size(); i++){
            Vector point = datapoints.get(i);
            if(point.x < x){
                datapoints.add(i, new Vector(x, y));
            }
        }
    }

    public void addDataPoint(Vector point){
        addDatapoint(point.x, point.y);
    }

    public enum DataMode{
        SCATTER,
        LINEAR,
        SQUARE
    }

    public enum SizeMode{
        ABSOLUTE,
        RELATIVE
    }
}

File: src/com/zypex/piflowapp/Graphics/Graph.kt
package com.zypex.piflowapp.Graphics

import javafx.scene.canvas.GraphicsContext

import javafx.scene.paint.Color
import com.zypex.piflow.math.*
import java.util.ArrayList

class Graph {
    private val datapoints: MutableList<Vector> = ArrayList()
    var color = Color.RED
    var dataMode = DataMode.LINEAR
    var sizeMode = SizeMode.RELATIVE
    fun drawSection(gc: GraphicsContext) {
        val w = gc.canvas.width
        val h = gc.canvas.height
    }

    fun addDatapoint(x: Double, y: Double) {
        for (i in datapoints.indices) {
            val point = datapoints[i]
            if (point.x < x) {
                datapoints.add(i, Vector(x, y))
            }
        }
    }

    fun addDataPoint(point: Vector) {
        addDatapoint(point.x, point.y)
    }

    enum class DataMode {
        SCATTER, LINEAR, SQUARE
    }

    enum class SizeMode {
        ABSOLUTE, RELATIVE
    }
}